### PR TITLE
Add ERA5 land forcing data for 2008

### PR DESCRIPTION
**Pull request by ph-kev**
Checklist:
- [X] I created a new folder `$artifact_name`
  - [X] I added a `README.md` in that that folder that
    - [X] describes the data and processing done to it
    - [X] lists the sources of the raw data
    - [X] lists the required citation, licenses
  - [X] If applicable (e.g., for Creative Commons), I added a `LICENSE` file
  - [X] I added the scripts that retrieve, process, and produce the artifact
  - [X] I added the environment used for such scripts (typically, `Project.toml`
        and `Manifest.toml`)
  - [x] I added the `OutputArtifacts.toml` file containing the information
        needed for package developers to add `$artifact_name` to their package
- [x] I uploaded the artifact folder to the Caltech cluster (in
      `/groups/esm/ClimaArtifacts/artifacts/$artifact_name`)
- [x] I added the relevant code to the `Overides.toml` on the Caltech Cluster
      (in `/groups/esm/ClimaArtifacts/artifacts/Overrides.toml`)
- [X] I added a link to the main `README.md` to point to the new artifact

Closes #5

### CODE REVIEW

**Comment by ph-kev** [comment]
Output of loading the dataset using NCDatasets.jl
```
Dataset: era5_2008_1.0x1.0.nc
Group: /

Dimensions
   time = 8784
   lon = 360
   lat = 181

Variables
  time   (8784)
    Datatype:    Dates.DateTime (Int32)
    Dimensions:  time
    Attributes:
     long_name            = time
     standard_name        = time
     units                = seconds since 1970-01-01
     calendar             = proleptic_gregorian

  lon   (360)
    Datatype:    Float32 (Float32)
    Dimensions:  lon
    Attributes:
     units                = degrees_east
     long_name            = longitude
     standard_name        = longitude

  lat   (181)
    Datatype:    Float32 (Float32)
    Dimensions:  lat
    Attributes:
     units                = degrees_north
     long_name            = latitude
     standard_name        = latitude

  lai_hv   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = m**2 m**-2
     missing_value        = 3.4028234663852886e38
     long_name            = Leaf area index, high vegetation
     standard_name        = unknown
     _FillValue           = NaN

  sp   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = Pa
     missing_value        = 3.4028234663852886e38
     long_name            = Surface pressure
     standard_name        = surface_air_pressure
     _FillValue           = NaN

  mtpr   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = kg m**-2 s**-1
     missing_value        = 3.4028234663852886e38
     long_name            = Mean total precipitation rate
     standard_name        = unknown
     _FillValue           = NaN

  d2m   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = K
     missing_value        = 3.4028234663852886e38
     long_name            = 2 metre dewpoint temperature
     standard_name        = unknown
     _FillValue           = NaN

  t2m   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = K
     missing_value        = 3.4028234663852886e38
     long_name            = 2 metre temperature
     standard_name        = unknown
     _FillValue           = NaN

  msdwswrf   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = W m**-2
     missing_value        = 3.4028234663852886e38
     long_name            = Mean surface downward short-wave radiation flux
     standard_name        = unknown
     _FillValue           = NaN

  u10   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = m s**-1
     missing_value        = 3.4028234663852886e38
     long_name            = 10 metre U wind component
     standard_name        = unknown
     _FillValue           = NaN

  lai_lv   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = m**2 m**-2
     missing_value        = 3.4028234663852886e38
     long_name            = Leaf area index, low vegetation
     standard_name        = unknown
     _FillValue           = NaN

  msr   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = kg m**-2 s**-1
     missing_value        = 3.4028234663852886e38
     long_name            = Mean snowfall rate
     standard_name        = unknown
     _FillValue           = NaN

  msdrswrf   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = W m**-2
     missing_value        = 3.4028234663852886e38
     long_name            = Mean surface direct short-wave radiation flux
     standard_name        = unknown
     _FillValue           = NaN

  v10   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = m s**-1
     missing_value        = 3.4028234663852886e38
     long_name            = 10 metre V wind component
     standard_name        = unknown
     _FillValue           = NaN

  msdwlwrf   (360 × 181 × 8784)
    Datatype:    Union{Missing, Float32} (Float32)
    Dimensions:  lon × lat × time
    Attributes:
     units                = W m**-2
     missing_value        = 3.4028234663852886e38
     long_name            = Mean surface downward long-wave radiation flux
     standard_name        = unknown
     _FillValue           = NaN
```

File: era5_land_forcing_data2008/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- Mean snowfall rate (lon, lat, time; "msr")
+- Mean surface direct short wave radiation flux (lon, lat, time; "msdrswrf")
+- Mean surface downward long wave radiation flux (lon, lat, time; "msdwlwrf")
+- Mean surface downward short wave radiation flux (lon, lat, time; "msdwswrf")
+- Mean total precipitation rate (lon, lat, time; "mtpr")
+- Leaf area index, high vegetation (lon, lat, time; "lai_hv")
+- Leaf area index, low vegetation (lon, lat, time; "lai_lv")
+
+## Prerequisites
+1. Python environment with cdsapi installed.
```
Comment (Sbozzolo) [inline era5_land_forcing_data2008/README.md:21]: Can you add a requirements.txt file or something that defines dependencies in a declarative way? You can see examples in some the land artifacts (@imreddyTeja can help)

File: era5_land_forcing_data2008/create_artifact.jl [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+include("thin_and_clean_artifact.jl")
+include("find_correct_order.jl")
+
+# Get all the files that end with .nc and sort them with respect to time
+file_paths = readdir()
+nc_paths = filter(file_path -> endswith(file_path, ".nc"), file_paths) |> sort
+nc_paths = find_correct_order(nc_paths)
+
+# Merge dataset across time dimension
+mfds = NCDataset(nc_paths, aggdim = "valid_time")
```
Comment (Sbozzolo) [inline era5_land_forcing_data2008/create_artifact.jl:13]: Can you close this dataset at the end of the script?

File: era5_land_forcing_data2008/create_artifact.jl [hunk 3]
```diff
@@ -0,0 +1,50 @@
+using ClimaArtifactsHelper
+using NCDatasets
+
+include("thin_and_clean_artifact.jl")
+include("find_correct_order.jl")
+
+# Get all the files that end with .nc and sort them with respect to time
+file_paths = readdir()
+nc_paths = filter(file_path -> endswith(file_path, ".nc"), file_paths) |> sort
+nc_paths = find_correct_order(nc_paths)
```
Comment (Sbozzolo) [inline era5_land_forcing_data2008/create_artifact.jl:10]: Can you add a check that there's the correct number of NetCDF files and instruct the user to run the python script if the files are not there?

File: era5_land_forcing_data2008/find_correct_order.jl [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+Takes in `fnames`, a list of file names of .nc files, and find the correct order
+to stitch the datasets together as one dataset along the time dimension.
+"""
+function find_correct_order(fnames)
+    first_days = []
+    for fname in fnames
+        nc = NCDataset(fname)
+        first_day = first(nc["valid_time"])
+        push!(first_days, first_day)
```
Comment (Sbozzolo) [inline era5_land_forcing_data2008/find_correct_order.jl:12]:
```suggestion
        NCDataset(fname) do nc
            first_day = first(nc["valid_time"])
            push!(first_days, first_day)
        end
```

**Review by Sbozzolo — APPROVED** [review]

File: era5_land_forcing_data2008/thin_and_clean_artifact.jl [hunk 5]
```diff
@@ -0,0 +1,87 @@
+"""
+   thin_and_clean_artifact(
+       filein,
+       fileout,
+       varname;
+       THINNING_FACTOR = 8,
+   )
+Take the file in `filein` and write a cleaned thinned-down version to `fileout`
+for the given `varname`. Thinning means taking one every `THINNING_FACTOR`
+points.
```
Comment (kmdeck) [inline era5_land_forcing_data2008/thin_and_clean_artifact.jl:10]: Could you also describe what "clean" means in this context?

File: era5_land_forcing_data2008/create_artifact.jl [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import OrderedCollections: OrderedDict
+
+include("thin_and_clean_artifact.jl")
+include("find_correct_order.jl")
+
+# Get all the files that end with .nc and sort them with respect to time
+file_paths = readdir()
+nc_paths = filter(file_path -> endswith(file_path, ".nc"), file_paths) |> sort
+
+# Replace this by checking all the dates instead
```
Comment (kmdeck) [inline era5_land_forcing_data2008/create_artifact.jl:12]: Does this still need to be here?

**Review by kmdeck — APPROVED** [review]
This looks great, thank you!

If I were to use the artifact locally, would it give a warning about not being able to download the 11GB file and then download the smaller file?

**Review by kmdeck — COMMENTED** [review]
One question before we merge, actually - this data has LAI at every hour, correct? We had discussed downloading it monthly as a separate file.

or perhaps we could thin it to be weekly and save it as another file?

File: era5_land_forcing_data2008/postprocess_and_make_weekly_lai_data.jl [hunk 7]
```diff
@@ -0,0 +1,95 @@
+"""
+   postprocess_and_make_weekly_lai_data(ncin, fileout)
+
+Take a `ncin` ,a .nc file loaded using NCDatasets, and write a postprocessed version to
```
Comment (Sbozzolo) [inline era5_land_forcing_data2008/postprocess_and_make_weekly_lai_data.jl:4]:
```suggestion
Take a `ncin`, a .nc file loaded using NCDatasets, and write a postprocessed version to
```

**Review by Sbozzolo — APPROVED** [review]
Looks good to me

File: era5_land_forcing_data2008/postprocess_and_make_weekly_lai_data.jl [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    defDim(ncout, "lon", Int(ceil(length(ncin["longitude"]))))
+    defDim(ncout, "lat", Int(ceil(length(ncin["latitude"]))))
+
+    time_ = defVar(
+        ncout,
+        "time",
+        Int32,
+        ("time",),
+        attrib = ncin["valid_time"].attrib,
+        deflatelevel = 9,
```
Comment (kmdeck) [inline era5_land_forcing_data2008/postprocess_and_make_weekly_lai_data.jl:44]: what is this?

**Review by kmdeck — APPROVED** [review]
This looks good - just a couple places to add a comment.

thanks very much for adding this!

**Comment by kmdeck** [comment]
@Sbozzolo can we close this now? https://github.com/CliMA/ClimaArtifacts/pull/5 (and delete the data from the cluster?, remove artifact from ClimaLand?)

**Comment by kmdeck** [comment]
sorry guys! I did not mean to close it.

@Sbozzolo can we close this now? https://github.com/CliMA/ClimaArtifacts/pull/5 (and delete the data from the cluster?, remove artifact from ClimaLand?)

File: era5_land_forcing_data2008/postprocess_and_make_weekly_lai_data.jl (hunk 8 above)
Comment (ph-kev) [reply era5_land_forcing_data2008/postprocess_and_make_weekly_lai_data.jl:44]: I am not sure what you are referring to, so I will assume that you are talking about `deflatelevel`. The parameter `deflatelevel` is the level of compression to the .nc file (higher number means more compression).

Comment (kmdeck) [reply era5_land_forcing_data2008/postprocess_and_make_weekly_lai_data.jl:44]: yes, exactly, I meant this line.

ok! I had not seen that before. thanks!

**Pull request merged**